cmd/web3/main.go

```diff
@@ -170,6 +170,35 @@ func main() {
 				GetAddressDetails(ctx, network, c.Args().First(), privateKey)
 			},
 		},
+		{
+			Name:  "balance",
+			Usage: "Retrieve token balance",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "address",
+					Usage: "Address to get balance for",
+				},
+				cli.StringFlag{
+					Name:  "contract",
+					Usage: "ERC-20 contract to use to get balance. Use either this or token flag",
+				},
+				cli.StringFlag{
+					Name:   "private-key,pk",
+					Usage:  "Private key",
+					EnvVar: pkVarName,
+				},
+				cli.StringFlag{
+					Name:  "token,coin",
+					Usage: "Token to check, must be one of the tokens supported in this tool",
+					Value: "GO",
+				},
+			},
+			Action: func(c *cli.Context) {
+				// Did this differently than the other, don't really know why we don't let the function have full access to the context
+				// That said, I don't like this double context thing either. @treeder
```

Comment:
It is unfortunate that they are both named 'Context', but they serve completely different purposes. In the beginning, we weren't using the `cli.Context` much, and were setting flags to store values into variables instead. I don't have strong feelings about using flag vars vs. using the context methods, but we should probably pick just one approach. One downside to passing the `cli.Context` through is that the function signatures become opaque, and you have to read the code or find usages to determine which parameters are used. Perhaps a clean in-between solution would be to use the `cli.Context` only directly from within the `Action` function, and pass the extracted values as parameters in to the function. For example:
```go
Action: func(c *cli.Context) {
  address := c.String("address")
  ...
  GetBalance(ctx, network, address, pk, token)
}
```
This also results in more easily testable functions by encapsulating the `cli.Context` access.